core-plugins/src/e2e-test/features/joiner/JoinerWithFile.feature

```diff
@@ -0,0 +1,259 @@
+    Then Navigate to the properties page of plugin: "File"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "joinerCsvNullFileFirstInput"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Click plugin property: "skipHeader"
+    Then Click plugin property: "enableQuotedValues"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "joinerCsvNullFileFirstSchema"
+    Then Validate "File2" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "File2"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "joinerCsvNullFileSecondInput"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Click plugin property: "skipHeader"
+    Then Click plugin property: "enableQuotedValues"
+    Then Click on the Get Schema button
+    Then Verify the Output Schema matches the Expected Schema: "joinerCsvNullFileSecondSchema"
+    Then Validate "File" plugin properties
+    Then Close the Plugin Properties page
+    When Navigate to the properties page of plugin: "Joiner"
+    Then Select joiner type "Inner"
+    Then Select radio button plugin property: "conditionType" with value: "basic"
+    Then Click plugin property: "joinerNullKeys"
+    Then Click on the Get Schema button
+    Then Validate "Joiner" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "File3"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "filePluginOutputFolder"
+    Then Replace input plugin property: "pathSuffix" with value: "yyyy-MM-dd-HH-mm-ss"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Click plugin property: "writeHeader"
+    Then Validate "File" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count of joiner is equal to IN record count of sink
+#    Enable below step once issue fixed (PLUGIN #1331)
+#    Then Validate output file generated by file sink plugin is equal to expected output file {string}
```

Comment:
Validation logic should be indepedent of the number of part files. You should read data from all the part files and verify that the data read matches the expectation

Reply:
Added the validation step.